workflows/sample/tasks/hifiasm_trio.wdl

```diff
@@ -0,0 +1,291 @@
+version 1.1
+
+#import "../../common/structs.wdl"
+
+import "https://raw.githubusercontent.com/ducatiMonster916/pb-human-wgs-workflow-wdl/main/workflows/common/structs.wdl"
+import "https://raw.githubusercontent.com/ducatiMonster916/pb-human-wgs-workflow-wdl/main/workflows/sample/tasks/hifiasm.wdl" as hifiasm
+
+
+task hifiasm_trio_assemble {
+  input {
+    Int threads = 48
+    String sample_name
+    String prefix = "~{sample_name}.asm"
+    String log_name = "hifiasm.log"
+
+    Array[File] movie_fasta
+    File parent1_yak
+    File parent2_yak
+    String pb_conda_image
+  }
+
+  Float multiplier = 2
+  Int disk_size = ceil(multiplier * size(movie_fasta, "GB")) + 20
+#  Int disk_size = 200
+  Int memory = threads * 3              #forces at least 3GB RAM/core, even if user overwrites threads
+
+  command <<<
+    echo requested disk_size =  ~{disk_size}
+    echo
+    source ~/.bashrc
+    conda activate hifiasm
+    echo "$(conda info)"
+
+    (hifiasm -o ~{prefix} -t ~{threads} -1 {parent1_yak} -2 {parent2_yak} ~{sep=" " movie_fasta} \
+    && (echo -e "hap1\t{params.parent1}\nhap2\t{params.parent2}" > ~{prefix}.key.txt) > ~{log_name} 2>&1
+  >>>
+  output {
+    File hap1_p_ctg        = "~{prefix}.bp.hap1.p_ctg.gfa"
+    File hap1_p_ctg_lowQ   = "~{prefix}.bp.hap1.p_ctg.lowQ.bed"
+    File hap1_p_noseq      = "~{prefix}.bp.hap1.p_ctg.noseq.gfa"
+    File hap2_p_ctg        = "~{prefix}.bp.hap2.p_ctg.gfa"
+    File hap2_p_ctg_lowQ   = "~{prefix}.bp.hap2.p_ctg.lowQ.bed"
+    File hap2_p_noseq      = "~{prefix}.bp.hap2.p_ctg.noseq.gfa"
+    File p_ctg             = "~{prefix}.bp.p_ctg.gfa"
+    File p_utg             = "~{prefix}.bp.p_utg.gfa"
+    File r_utg             = "~{prefix}.bp.r_utg.gfa"
+    File ec_bin            = "~{prefix}.ec.bin"
+    File ovlp_rev_bin      = "~{prefix}.ovlp.reverse.bin"
+    File ovlp_src_bin      = "~{prefix}.ovlp.source.bin"
```

Comment:
There's no reason to keep these outputs (*.p_utg.gfa, *.r_utg.gfa, `*.bin`).  We also might not need to keep `~{prefix}.bp.p_ctg.gfa`.  We should check with CMKC on that.

Also, I'm not sure how straightforward this is, but we don't need to keep these after running gfa2fasta:
`{prefix}.bp.hap1.p_ctg.gfa`
`{prefix}.bp.hap2.p_ctg.gfa`

Reply:
So because we've designed this for cloud native implementation- by default we save the pointers to all intermediate files in the output.json (that's what's actually why we put anything in the "output" section of the WDL)- it becomes very difficult to find those files otherwise. 

CMH has data retention policies in place that will automatically purge intermediate file results and will leverage CROO (to copy specific files they want to destination with specified naming schema). any intermediate files will not be gathered and purged automatically.

Reply:
Agreed this is mostly for a pointer and we can use CROO and our implementation for users to decided what they want to retain.